graph/db/sql_store.go

```diff
@@ -5082,3 +5050,54 @@ func batchLoadChannelPolicyExtrasHelper(ctx context.Context,
 		},
 	)
 }
+
+// forEachNodePaginated executes a paginated query to process each node in the
+// graph. It uses the provided SQLQueries interface to fetch nodes in batches
+// and applies the provided processNode function to each node.
+func forEachNodePaginated(ctx context.Context, cfg *sqldb.QueryConfig,
+	db SQLQueries, processNode func(context.Context, int64,
+		*models.LightningNode) error) error {
+
+	pageQueryFunc := func(ctx context.Context, lastID int64,
+		limit int32) ([]sqlc.GraphNode, error) {
+
+		return db.ListNodesPaginated(
+			ctx, sqlc.ListNodesPaginatedParams{
+				Version: int16(ProtocolV1),
```

Comment:
Will the content of this function change with v2? If not, perhaps we can pass in the protocol version as a parameter to make this function re-usable for v2 as well.

Reply:
good idea :)